baby-bear/src/baby_bear.rs

```diff
@@ -402,6 +396,13 @@ fn to_monty_64(x: u64) -> u32 {
     (((x as u128) << MONTY_BITS) % P as u128) as u32
 }
 
+#[inline]
+#[must_use]
+// This is almost certainly not the fastest implementation and should be improved to not use % twice.
+fn to_monty_128(x: u128) -> u32 {
```

Comment:
I think we never use this function so it probably doesn't matter but definitely feels like there must be a better implementation.